src/pytorch_lightning/strategies/single_tpu.py

```diff
@@ -55,13 +55,10 @@ def is_distributed(self) -> bool:
         return False
 
     def setup(self, trainer: "pl.Trainer") -> None:
+        assert self.model, "self.model must be set before find_shared_parameters(self.model)"
         shared_params = find_shared_parameters(self.model)
         self.model_to_device()
-        if is_overridden("on_post_move_to_device", self.lightning_module):
-            self.model.on_post_move_to_device()
-        else:
-            set_shared_parameters(self.model, shared_params)
```

Comment:
why did you remove this logic? Never change existing logic for the sake of typing!

Reply:
Sure I correct.

Put aside the fact that I don't have to do it in this PR, I changed it because as explained in the PR, It appeared to me that this ```if``` cannot be ```True```